src/main/java/dci/j24e01/OurShop/services/ProductDAOImpl.java

```diff
@@ -47,7 +47,29 @@ public List<Product> getProducts() {
             return null;
         }
     }
+    public List<Product> getProductsPaginated(int page, int size) {
```

Comment:
as stated before, this could be an @Override if the methods where defined in the DAO

Reply:
I fixed it